pkg/appmgmt/sparkoperator/spark.go

```diff
@@ -124,3 +127,21 @@ func (os *Manager) deleteApplication(obj interface{}) {
 	log.Logger().Info("spark app deleted", zap.Any("SparkApplication", app))
 	os.amProtocol.NotifyApplicationComplete(app.Status.SparkApplicationID)
 }
+
+func GetProxyUser(pod *v1.Pod) string {
+	if appName, ok := pod.Labels[constants.SparkOperatorLabelAppName]; ok {
+		if crdInformerFactory == nil {
+			log.Logger().Info("Spark operator AppMgmt service is not initialized, so the username from the SparkApp CRD cannot be obtained")
+			return ""
+		}
+		app, err := crdInformerFactory.Sparkoperator().V1beta2().SparkApplications().Lister().SparkApplications(pod.Namespace).Get(appName)
```

Comment:
This should not make an API server call but use the informers instead. Basically, cache all the Spark application objects in-memory (track add/update/delete events) and use those to lookup the proxy user.